Emby.Server.Implementations/Session/SessionManager.cs

```diff
@@ -108,9 +91,49 @@ public class SessionManager : ISessionManager, IDisposable
             _authRepo = authRepo;
             _deviceManager = deviceManager;
             _mediaSourceManager = mediaSourceManager;
+
             _deviceManager.DeviceOptionsUpdated += OnDeviceManagerDeviceOptionsUpdated;
         }
 
+        /// <inheritdoc />
+        public event EventHandler<GenericEventArgs<AuthenticationRequest>> AuthenticationFailed;
+
+        /// <inheritdoc />
+        public event EventHandler<GenericEventArgs<AuthenticationResult>> AuthenticationSucceeded;
+
+        /// <summary>
+        /// Occurs when [playback start].
+        /// </summary>
+        public event EventHandler<PlaybackProgressEventArgs> PlaybackStart;
+
+        /// <summary>
+        /// Occurs when [playback progress].
+        /// </summary>
+        public event EventHandler<PlaybackProgressEventArgs> PlaybackProgress;
+
+        /// <summary>
+        /// Occurs when [playback stopped].
+        /// </summary>
+        public event EventHandler<PlaybackStopEventArgs> PlaybackStopped;
```

Comment:
Not a fan of the brackets. I also noticed `PlaybackStart` and `PlaybackStopped` have different tenses, we should probably fix that at some point.